File: library/src/main/java/com/nasahapps/nasahutils/app/BaseActivity.java
package com.nasahapps.nasahutils.app;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;

/**
 * Created by Hakeem on 7/26/15.
 */
public abstract class BaseActivity extends AppCompatActivity {

    protected static String TAG;

    protected Menu mMenu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TAG = getClass().getSimpleName();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        mMenu = menu;
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
        }

        return super.onOptionsItemSelected(item);
    }
}

File: library/src/main/java/com/nasahapps/nasahutils/app/BaseActivity.kt
package com.nasahapps.nasahutils.app

import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem

/**
 * Created by Hakeem on 7/26/15.
 */
abstract class BaseActivity : AppCompatActivity() {

    protected val TAG = javaClass.simpleName
    protected var menu: Menu? = null

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        this.menu = menu
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
        }

        return super.onOptionsItemSelected(item)
    }
}
